examples/connext_dds/dynamic_data_request_reply/cpp/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# (c) 2020 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+# RTI grants Licensee a license to use, modify, compile, and create derivative
+# works of the Software.  Licensee has the right to distribute object form only
+# for use with RTI products.  The Software is provided "as is", with no warranty
+# of any type, including any warranty for fitness for any purpose. RTI is under
+# no obligation to maintain or support the Software.  RTI shall not be liable for
+# any incidental or consequential damages arising out of the use or inability to
+# use the software.
+
+cmake_minimum_required(VERSION 3.7)
+project(rtiexamples-dynamic-data-request_reply)
+set(CMAKE_MODULE_PATH
+    ${CMAKE_MODULE_PATH}
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../../resources/cmake"
+)
+
+# Find the RTI Connext DDS libraries
+if(NOT RTIConnextDDS_FOUND)
+    find_package(RTIConnextDDS
+        "5.3.0"
+        REQUIRED
+        COMPONENTS
+            core
+            messaging_api
+    )
+endif()
+
+
+
+add_executable(prime_number_requester_cxx
+    "${CMAKE_CURRENT_SOURCE_DIR}/PrimeNumberRequester.cxx"
+)
+
+add_executable(prime_number_replier_cxx
+    "${CMAKE_CURRENT_SOURCE_DIR}/PrimeNumberReplier.cxx"
+)
+
+
+
+#We add the library:
+add_library(PrimesType "${CMAKE_CURRENT_SOURCE_DIR}/PrimesType.cxx")
+
+target_link_libraries(PrimesType
+    PRIVATE
+        RTIConnextDDS::cpp_api
```

Comment:
When you add the `RTIConnextDDS::messaging_cpp_api` target, the `RTIConnextDDS::cpp_api` target is included automatically. So, you don't need to link against `RTIConnextDDS::cpp_api`